### ISSUE TITLE
Inconsistent torch complex after STFT

### ISSUE THREAD

**Issue by nhanerc** [pytorch/pytorch#122780]
### 🐛 Describe the bug

```python
import torch

x = torch.randn(1, 16000).float()
y1 = torch.stft(x, n_fft=256, hop_length=160, return_complex=False)
y2 = torch.stft(x, n_fft=256, hop_length=160, return_complex=True)
y3 = torch.view_as_real(y2)

y1 = torch.sqrt(torch.sum(y1 ** 2, dim=-1))
y2 = y2.abs()
y3 = torch.sqrt(torch.sum(y3 ** 2, dim=-1))

print(torch.max(torch.abs(y2 - y1)))
print(torch.max(torch.abs(y3 - y1)))
```
Sample output:
```bash
tensor(3.8147e-06)
tensor(0.)
```

I know the difference is small, but the error gain overtime is big. I expect that the first difference is zero as well.
Does anyone know something wrong?

### Versions

```
[pip3] numpy==1.24.4
[pip3] onnx==1.16.0
[pip3] onnxruntime==1.17.1
[pip3] torch==2.2.1
[pip3] torchvision==0.17.1
[pip3] triton==2.2.0
[conda] numpy                     1.24.4                   pypi_0    pypi
[conda] torch                     2.2.1                    pypi_0    pypi
[conda] torchvision               0.17.1                   pypi_0    pypi
[conda] triton                    2.2.0                    pypi_0    pypi
```

**Comment by lezcano** [comment]
Alas this is expected. Not only here but in pretty much any function, including functions as simple as `torch.sum`.
See https://pytorch.org/docs/stable/notes/numerical_accuracy.html

Unrelated, consider using `torch.linalg.vector_norm` to compute the 2-norm faster.

**Comment by lezcano** [comment]
Actually, the small error is in the way you compute the norm. The functions return identical results:

```python
import torch

x = torch.randn(1, 16000).float()
y1 = torch.stft(x, n_fft=256, hop_length=160, return_complex=False)
y2 = torch.stft(x, n_fft=256, hop_length=160, return_complex=True)
y3 = torch.view_as_real(y2)
print(torch.max(torch.abs(y3 - y1)))
```
returns 0

**Comment by nhanerc** [comment]
Thanks for replying to this issue.

I'm quite surprised that the normal from `y2.abs()` is different from `torch.sqrt(torch.sum(torch.view_as_real(y2) ** 2, dim=-1))`.

**Comment by lezcano** [comment]
This is expected as per https://pytorch.org/docs/stable/notes/numerical_accuracy.html. If you want more accuracy, you can perform your computations on `float64` at the expense of a big perf hit.
